mockdockerclient/image_api_client.go

```diff
@@ -83,10 +84,13 @@ func (c *ImageAPIClient) ImageLoad(context.Context, io.Reader, bool) (types.Imag
 }
 
 // ImagePull is a mock implementation of Docker's client.ImageAPIClient.ImagePull()
-func (c *ImageAPIClient) ImagePull(context.Context, string, types.ImagePullOptions) (io.ReadCloser, error) {
+func (c *ImageAPIClient) ImagePull(ctx context.Context, ref string, opts types.ImagePullOptions) (io.ReadCloser, error) {
 	if c.PullResp == nil {
 		return nil, Err
 	}
+	if c.Platform != opts.Platform {
```

Comment:
Don't perform this check. The mock is designed to be simple and not contain any logic so the test consumer can easily setup test scenarios.